src/Unity.Mathematics/math.cs

```diff
@@ -6728,27 +6740,28 @@ public static unsafe uint hash(void* pBuffer, int numBytes, uint seed = 0)
                     int count = numBytes >> 4;
                     for (int i = 0; i < count; ++i)
                     {
-                        state += *p++ * Prime2;
-                        state = (state << 13) | (state >> 19);
+                        var data = new uint4(read32_little_endian(p), read32_little_endian(p + 4), read32_little_endian(p + 8), read32_little_endian(p + 12));
```

Comment:
This is the fix. Read little endian byte by byte every time instead of assuming `uint`.